pallets/parachain-staking/src/lib.rs

```diff
@@ -507,13 +759,19 @@ pub mod pallet {
 
 	#[pallet::hooks]
 	impl<T: Config> Hooks<BlockNumberFor<T>> for Pallet<T> {
-		// This upgrade fixes a bug that may have led to an incorrect `Total` (total staked)
 		fn on_runtime_upgrade() -> Weight {
+			// migrate from Collator -> Collator2
+			for (acc, collator_state) in CollatorState::<T>::drain() {
+				let state: Collator2<T::AccountId, BalanceOf<T>> = collator_state.into();
+				<CollatorState2<T>>::insert(acc, state);
+			}
+
+			// correct any incorrectly set `Total`
 			let old_total = Total::<T>::get();
 			let mut new_total: BalanceOf<T> = 0u32.into();
 
-			for collator_state in CollatorState::<T>::iter_values() {
-				new_total += collator_state.total;
+			for collator_state in CollatorState2::<T>::iter_values() {
+				new_total += collator_state.total_backing;
 			}
 
 			Total::<T>::put(new_total);
```

Comment:
This is the last thing I'm wondering about. This function now does two orthogonal migrations. I don't necessarily have a problem with it, I'm just hesitant to merge until @crystalin, who actually runs the sudo upgrades on alphanet, is on the same page about how / when we'll deploy these migrations.

I guess the migration about resetting the `Total`s is harmless except for execution time. It wouldn't corrupt any data to run it twice.

Reply:
> I guess the migration about resetting the Totals is harmless except for execution time. It wouldn't corrupt any data to run it twice.

Agreed, we really only need to ensure that this migration only runs once on every network. We could add a storage bool that is default false and then we set it to true if this runs so it doesn't run twice.